### DISCUSSION TITLE
Statistics for each load simulations

### DISCUSSION THREAD

**Discussion by fabiocbr75** [VIP-Logic/NBomber#516] [Q&A]
Hello,
I have the same scenario with different load simulations.
At the end of the tests, the result is only one for the scenario but I need to have the statistics splitted between simulations.
Below my code snippet:

```
var scenario1 = ScenarioBuilder
                               .CreateScenario("scenario1", steps.ToArray())
                               .WithoutWarmUp()
                               .WithLoadSimulations(
                                    Simulation.KeepConstant(copies: 1, during: TimeSpan.FromSeconds(5)),
                                    Simulation.KeepConstant(copies: 3, during: TimeSpan.FromSeconds(5))
                                );
```
Is it possible to have the statistics in a single report?

Thanks

**Comment by AntyaDev** [comment]
Hi @fabiocbr75 
First, please try to use v4-beta since it will be released very soon.

Do you want to run the same scenario but with different load simulations and get separated statistics?

**Comment by fabiocbr75** [comment]
Hi,
>Do you want to run the same scenario but with different load simulations and get separated statistics?

Yes, it's my dream :D

Until now I used version 3.3.
I'll try to use the v4-beta in the next days.

Thanks

**Reply by AntyaDev** [reply]
I guess that some API changes should be introduced to support it.
Can you suggest what API you expect to configure such an example?

**Comment by fabiocbr75** [comment]
Hello,

Only add to the current overall statistics results the statistics for each simulation in the generated reports.
Also, add every scenario execution raw data in some txt/csv file. Now the scenario simulation executions are embedded only in the HTML file for the chart visualization.
I check and the NBomberRunner Run, returns the ScenarioStats.... it is better to have the LoadSimulationStats as an array for the simulation results filled with all simulations with the related statistics result plus an "overall" simulation for the aggregated result. 

Thanks

**Reply by AntyaDev** [reply]
not sure that I understood
